src/redux/api/templateApi.ts

```diff
@@ -0,0 +1,66 @@
+import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
+import { environment } from "@/constants/environment";
+import { ITemplate, ITemplateDataDb } from "@/interfaces/template";
+import _ from "lodash";
+
+export const templateApi = createApi({
+  reducerPath: "templates",
+  baseQuery: fetchBaseQuery({
+    baseUrl: environment.apiBaseUrl,
+  }),
+  tagTypes: ["template"],
+  endpoints: (builder) => ({
+    getTemplates: builder.query<Omit<ITemplateDataDb, "__v" | "isDeleted">[], string>({
+      query: (userId) => `/templates?user_id=${userId}`,
+      transformResponse: (result: ITemplateDataDb[], _meta, _arg) => {
+        return result.map((item) => _.omit(item, ["__v", "isDeleted"]));
+      },
+      providesTags: (result, _err, _arg) =>
+        result
+          ? [...result.map(({ _id }) => ({ type: "template" as const, id: _id })), "template"]
+          : ["template"],
+    }),
```

Comment:
Same here, it is my suggestion that you can add {type:"template", id:"LIST"} after { type: "template", id: arg._id }